### PR TITLE
web: Upgrade to Finatra 1.1.1

### PR DESCRIPTION
**Pull request by franklinhu**

### CODE REVIEW

File: zipkin-web/src/main/scala/com/twitter/zipkin/web/ZipkinWeb.scala [hunk 1]
```diff
@@ -35,36 +34,19 @@ class ZipkinWeb(
...
-        .codec(Http())
-        .bindTo(new InetSocketAddress(serverPort))
-        .name("ZipkinWeb")
-        .tracerFactory(tracerFactory)
-        .build(service)
-    }
-    log.info("Finatra service started in port: " + serverPort)
-  }
+    /* Finatra uses System properties for configuration */
+    System.setProperty("name", "ZipkinWeb")
```
Comment (johanoskarsson) [inline zipkin-web/src/main/scala/com/twitter/zipkin/web/ZipkinWeb.scala]: Wow, this feels very wrong.

File: zipkin-web/src/main/scala/com/twitter/zipkin/web/ZipkinWeb.scala [hunk 2]
```diff
@@ -35,36 +34,19 @@ class ZipkinWeb(
...
-
-    server = Some {
-      ServerBuilder()
-        .codec(Http())
-        .bindTo(new InetSocketAddress(serverPort))
-        .name("ZipkinWeb")
-        .tracerFactory(tracerFactory)
-        .build(service)
-    }
-    log.info("Finatra service started in port: " + serverPort)
```
Comment (johanoskarsson) [inline zipkin-web/src/main/scala/com/twitter/zipkin/web/ZipkinWeb.scala:58]: Would be nice to keep this

File: zipkin-web/src/main/scala/com/twitter/zipkin/web/ZipkinWeb.scala [hunk 3]
```diff
@@ -35,36 +34,19 @@ class ZipkinWeb(
...
+    s.start(tracerFactory)
 
-  def shutdown() {
-    server.foreach { _.close() }
   }
 
-  def register(app: AbstractFinatraController[Request, Future[Response], Future[HttpResponse]]) {
-    controllers.add(app)
-  }
+  def shutdown() {}
```
Comment (johanoskarsson) [inline zipkin-web/src/main/scala/com/twitter/zipkin/web/ZipkinWeb.scala]: How does it shut down now?

File: zipkin-web/src/main/scala/com/twitter/zipkin/web/ZipkinWeb.scala (hunk 1 above)
Comment (franklinhu) [inline zipkin-web/src/main/scala/com/twitter/zipkin/web/ZipkinWeb.scala]: Finatra does config through system properties right now. It'd be a stopgap until @capotej gets out a more robust configuration system.

**Comment by franklinhu** [comment]
@johanoskarsson We don't necessarily have to use the `FinatraServer` class, could just keep sort of what we had before and piece it together

**Comment by franklinhu** [comment]
@johanoskarsson This feels much cleaner now.

**Comment by johanoskarsson** [comment]
+1

**Comment by capotej** [comment]
woot!

Yeah, I'm looking to replace my hacky config system with something cleaner in the future, perhaps https://github.com/typesafehub/config

**Pull request closed without merging**